### PR TITLE
Prettier & SwiftLint

### PR DESCRIPTION
**Pull request by dwieeb**
This PR adds [prettier](https://prettier.io/) and [SwiftLint](https://github.com/realm/SwiftLint) to Capacitor. I'm not committing formatted files yet to avoid conflicts.

- HTML, JS, CSS, and TS are supported by prettier out of the box.
- For Java, there's [prettier-plugin-java](https://github.com/jhipster/prettier-java).
- For Swift, there's [SwiftLint](https://github.com/realm/SwiftLint).

To see how prettier and SwiftLint format source files:
1. Pull down the branch
1. ~Install SwiftLint: `brew install swiftlint`~
1. Run `npm run fmt` in the root of the repo. Prettier is just a style formatter, so it should format everything.
1. SwiftLint can't autofix everything, so run `npx swiftlint` in the root of repo to see what it complains about.

**TODO**
- [ ] Commit formatted and linted files.
- [ ] Setup CI to run prettier and SwiftLint

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,10 +2,22 @@
...
+    "lint": "npm run prettier -- --check && npm run swiftlint -- lint",
+    "fmt": "npm run prettier -- --write && npm run swiftlint -- autocorrect",
+    "prettier": "prettier '+(app-template/**/*.{css,html,js}|{cli,core,plugin-template}/src/**/*.{css,html,js,ts}|{android/capacitor,android-template/app,capacitor-cordova-android-plugins}/src/**/*.java)'",
+    "swiftlint": "swiftlint",
     "publish": "bash scripts/prerelease.sh && lerna run build && lerna publish --force-publish --skip-git && npm run publish-push",
     "publish-push": "bash scripts/deploy.sh"
   },
+  "husky": {
+    "hooks": {
+      "pre-commit": "npm run lint"
```
Comment (smpeters) [inline package.json:14]: IllustriousDownrightGrasshopper-size_restricted

**Comment by jcesarmobile** [comment]
just noticed the prettier command doesn't work on windows
nor any other of the new commands since swiftlint can't be used on windows

The contributing file says 
> Install [SwiftLint](https://github.com/realm/SwiftLint) if you plan to contribute to iOS code. 
but at the moment it's mandatory even if you don't edit iOS code since the pre-commit hook will run swiftlint and fail if not installed

**Comment by dwieeb** [comment]
Now we are using this: https://github.com/ionic-team/swiftlint

This package provides a wrapper around swiftlint that doesn't just prints a message for Windows. Also, lint tools should be installed locally in each project, anyway, so I removed the brew install instructions.

**Comment by jcesarmobile** [comment]
the prettier command still fails on windows, doesn't seem to understand the `|{cli` part

**Comment by dwieeb** [comment]
Now using the new [`@ionic/prettier-config`](https://github.com/ionic-team/prettier-config). The plugin template has also [been updated](https://github.com/ionic-team/capacitor/pull/3241) to use it.

**Comment by dwieeb** [comment]
Same for [`@ionic/swiftlint-config`](https://github.com/ionic-team/swiftlint-config/).

**Pull request merged**